includes/user.php

```diff
@@ -227,6 +227,18 @@ public static function is_user_notice_viewed( $notice_id ) {
 		return $notices[ $notice_id ]['is_viewed'] ?? false;
 	}
 
+	/**
+	 * Checks whether the current user is allowed to upload JSON files.
+	 *
+	 * Note: The 'json-upload' capability is managed by the Role Manager as a part of its blacklist restrictions.
+	 * In this context, we are negating the user's permission check to use it as a whitelist, allowing uploads.
+	 *
+	 * @return bool Whether the current user can upload JSON files.
+	 */
+	public static function is_current_user_can_upload_json() {
+		return current_user_can( 'manage_options' ) || ! Plugin::instance()->role_manager->user_can( 'json-upload' );
+	}
```

Comment:
It feels like duplication..

Reply:
No, admins will have the capability by default. Other roles - only if specifically given the permission in Role Editor.
role_manager->user_can() does not rely on WP user_can(), but is a custom one, that checks specifically the permissions set in the Role Editor.